usr/share/rear/build/default/980_verify_rootfs.sh

```diff
@@ -1,18 +1,63 @@
+local rootfs_dir_fs_mountpoint=$( filesystem_name $ROOTFS_DIR )
+if grep -qE '^\S+ '$rootfs_dir_fs_mountpoint' \S+ \S*\bnoexec\b\S* ' /proc/mounts ; then
+    # Intentionally failing here with Error to make it mandatory for ReaR to really check the rescue image.
+    # It is much more important to guarantee a runnable rescue image than to support noexec environments,
+    # cf. https://github.com/rear/rear/pull/1514#discussion_r140752346
+    Error "Cannot test if the ReaR recovery system is usable because $rootfs_dir_fs_mountpoint is mounted 'noexec'"
+fi
+Log "Testing that $ROOTFS_DIR contains a usable system"
 
-if grep -qE '^\S+ '$tmp_filesystem' \S+ \S*\bnoexec\b\S* ' /proc/mounts; then
-	LogPrint "WARNING: Filesystem $tmp_filesystem is mounted 'noexec', aborting chroot bash test"
-	Log "The above error means that we cannot test our chrooted environment because
-we need the 'exec' option set to the filesystem. One way to achieve this is
-by doing: mount -o remount,exec $tmp_filesystem"
-	return
+# The bash test ensures that we have a working bash in the ReaR recovery system:
+if ! chroot $ROOTFS_DIR /bin/bash -c true ; then
+    KEEP_BUILD_DIR=1
+    BugError "ReaR recovery system in '$ROOTFS_DIR' is broken: 'bash -c true' failed"
 fi
 
-# The chroot bash test ensures that we have a working bash on our rescue image
-if ! chroot $ROOTFS_DIR bash -c true ; then
-	KEEP_BUILD_DIR=1
-	BugError "ROOTFS_DIR '$ROOTFS_DIR' is broken, chroot bash test failed."
+# The ldd test ensures that for dynamically linked executables the required libraries are there.
+# The ldd test runs after the bash test because /bin/ldd is a bash script.
+# First test is 'ldd /bin/bash' to ensure 'ldd' works:
+Log "Testing 'ldd /bin/bash' to ensure 'ldd' works for the subsequent 'ldd' tests"
+if ! chroot $ROOTFS_DIR /bin/ldd /bin/bash 1>&2 ; then
+    KEEP_BUILD_DIR=1
+    BugError "ReaR recovery system in '$ROOTFS_DIR' is broken: 'ldd /bin/bash' failed"
+fi
+# Now test each binary (except links) with ldd and look for 'not found' libraries.
+# In case of 'not found' libraries for dynamically linked executables ldd returns zero exit code.
+# When running ldd for a file that is 'not a dynamic executable' ldd returns non-zero exit code.
+local binary=""
+local broken_binaries=""
+# Catch all binaries and libraries also e.g. those that are copied via COPY_AS_IS into other paths:
+for binary in $( find $ROOTFS_DIR -type f -executable -printf '/%P\n' ) ; do
+    chroot $ROOTFS_DIR /bin/ldd $binary | grep -q 'not found' && broken_binaries="$broken_binaries $binary"
+done
+if contains_visible_char "$broken_binaries" ; then
+    KEEP_BUILD_DIR=1
+    LogPrint "Checking for binaries or libraries in the ReaR recovery system that require additional libraries"
+    local fatal_missing_library=""
+    local ldd_output=""
+    for binary in $broken_binaries ; do
+        if grep -q '^/bin/' <<<"$binary" ; then
```

Comment:
Why not check for `/bin/` instead to catch at least those binaries copied by `COPY_AS_IS` into other `bin` paths?

Reply:
I use now '/bin/'